var/spack/repos/builtin/packages/perl/package.py

```diff
@@ -143,8 +156,8 @@ def setup_dependent_environment(self, spack_env, run_env, dependent_spec):
         for d in dependent_spec.traverse(
                 deptype=('build', 'run'), deptype_query='run'):
             if d.package.extends(self.spec):
-                perl_lib_dirs.append(join_path(d.prefix, 'lib', 'perl5'))
-                perl_bin_dirs.append(join_path(d.prefix, 'bin'))
+                perl_lib_dirs.append(join_path(d.prefix.lib, 'perl5'))
```

Comment:
I have updated to the new path syntax in 12e02ece6f6b76d374d16ee552aff4d4ab03e88d. This also required a rebase onto a recent develop branch.